# Упражнения по NumPy

#### 1. Импортируйте NumPy as np

In [2]:
import numpy as np

#### 2. Создайте массив из 10 нулей

In [3]:
# Ваше решение тут
a1 = np.zeros(10)
a1

array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])

Результат:

<code>array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])</code>

#### 3. Создайте матрицу 3 на 4 из единиц

In [6]:
# Ваше решение тут
a2 = np.ones((3, 4))
a2

array([[1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.]])

Результат:

<code>array([[1., 1., 1., 1.],
        [1., 1., 1., 1.],
        [1., 1., 1., 1.]])</code>

#### 4. Создайте матрицу 5 на 5 из пятёрок

In [9]:
# Ваше решение тутa
a3 = np.full((5, 5), 5)
a3

array([[5, 5, 5, 5, 5],
       [5, 5, 5, 5, 5],
       [5, 5, 5, 5, 5],
       [5, 5, 5, 5, 5],
       [5, 5, 5, 5, 5]])

Результат:

<code>array([[5, 5, 5, 5, 5],
        [5, 5, 5, 5, 5],
        [5, 5, 5, 5, 5],
        [5, 5, 5, 5, 5],
        [5, 5, 5, 5, 5]])</code>

#### 5. Создайте массив целых чисел от 10 до 50 включительно

In [16]:
# Ваше решение тут
a4 = np.arange(10, 51)
a4

array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
       27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43,
       44, 45, 46, 47, 48, 49, 50])

Результат:

<code>array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50])</code>

#### 6. Создайте массив всех чётных чисел от 10 до 50 включительно

In [17]:
# Ваше решение тут
a5 = np.arange(10, 51, 2)
a5

array([10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42,
       44, 46, 48, 50])

Результат:

<code>array([10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50])</code>

#### 7. Создайте матрицу 6 на 10 со значениями от 1 до 60 включительно

In [20]:
# Ваше решение тут
a6 = np.arange(1, 61)
a6.reshape(6, 10)

array([[ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10],
       [11, 12, 13, 14, 15, 16, 17, 18, 19, 20],
       [21, 22, 23, 24, 25, 26, 27, 28, 29, 30],
       [31, 32, 33, 34, 35, 36, 37, 38, 39, 40],
       [41, 42, 43, 44, 45, 46, 47, 48, 49, 50],
       [51, 52, 53, 54, 55, 56, 57, 58, 59, 60]])

Результат:

<code>array([[ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10],
        [11, 12, 13, 14, 15, 16, 17, 18, 19, 20],
        [21, 22, 23, 24, 25, 26, 27, 28, 29, 30],
        [31, 32, 33, 34, 35, 36, 37, 38, 39, 40],
        [41, 42, 43, 44, 45, 46, 47, 48, 49, 50],
        [51, 52, 53, 54, 55, 56, 57, 58, 59, 60]])</code>

#### 8. Создайте единичную матрицу размером 3 на 3

In [22]:
# Ваше решение тут
a7 = np.eye(3, 3)
a7

array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

Результат:

<code>array([[1., 0., 0.],
        [0., 1., 0.],
        [0., 0., 1.]])</code>

#### 9. С помощью NumPy сгенерируйте матрицу 3 на 2 из целых чисел в диапазоне от 1 до 100 включительно

In [25]:
np.random.seed(0)
# Ваше решение тут
a8 = np.random.randint(1, 100, (3, 2))
a8

array([[45, 48],
       [65, 68],
       [68, 10]], dtype=int32)

Результат:

<code>array([[45, 48],
        [65, 68],
        [68, 10]], dtype=int32)</code>

#### 10. С помощью NumPy сгенерируйте массив из 25 случайных чисел, выбранных из стандартного нормального распределения

In [30]:
np.random.seed(0)
# Ваше решение тут
np.random.randn(25)

array([ 1.76405235,  0.40015721,  0.97873798,  2.2408932 ,  1.86755799,
       -0.97727788,  0.95008842, -0.15135721, -0.10321885,  0.4105985 ,
        0.14404357,  1.45427351,  0.76103773,  0.12167502,  0.44386323,
        0.33367433,  1.49407907, -0.20515826,  0.3130677 , -0.85409574,
       -2.55298982,  0.6536186 ,  0.8644362 , -0.74216502,  2.26975462])

Результат:

<code>array([1.76405235, 0.40015721, 0.97873798, 2.2408932, 1.86755799, -0.97727788, 0.95008842, -0.15135721, -0.10321885, 0.4105985, 0.14404357, 1.45427351, 0.76103773, 0.12167502, 0.44386323, 0.33367433, 1.49407907, -0.20515826, 0.3130677 , -0.85409574, -2.55298982, 0.6536186, 0.8644362, -0.74216502, 2.26975462])</code>

#### 11. Создайте массив из 15 равноудалённых друг от друга точек между 1 и 10:

In [34]:
# Ваше решение тут
np.linspace(1, 10, 15)


array([ 1.        ,  1.64285714,  2.28571429,  2.92857143,  3.57142857,
        4.21428571,  4.85714286,  5.5       ,  6.14285714,  6.78571429,
        7.42857143,  8.07142857,  8.71428571,  9.35714286, 10.        ])

Результат:

<code>array([1. , 1.64285714, 2.28571429, 2.92857143, 3.57142857, 4.21428571, 4.85714286, 5.5, 6.14285714, 6.78571429, 7.42857143, 8.07142857, 8.71428571, 9.35714286, 10.])</code>

#### 12. Создайте указанную ниже матрицу:

In [42]:
# Ваше решение тут
np.arange(0.01, 1, 0.01)

array([0.01, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09, 0.1 , 0.11,
       0.12, 0.13, 0.14, 0.15, 0.16, 0.17, 0.18, 0.19, 0.2 , 0.21, 0.22,
       0.23, 0.24, 0.25, 0.26, 0.27, 0.28, 0.29, 0.3 , 0.31, 0.32, 0.33,
       0.34, 0.35, 0.36, 0.37, 0.38, 0.39, 0.4 , 0.41, 0.42, 0.43, 0.44,
       0.45, 0.46, 0.47, 0.48, 0.49, 0.5 , 0.51, 0.52, 0.53, 0.54, 0.55,
       0.56, 0.57, 0.58, 0.59, 0.6 , 0.61, 0.62, 0.63, 0.64, 0.65, 0.66,
       0.67, 0.68, 0.69, 0.7 , 0.71, 0.72, 0.73, 0.74, 0.75, 0.76, 0.77,
       0.78, 0.79, 0.8 , 0.81, 0.82, 0.83, 0.84, 0.85, 0.86, 0.87, 0.88,
       0.89, 0.9 , 0.91, 0.92, 0.93, 0.94, 0.95, 0.96, 0.97, 0.98, 0.99])

Результат:

<code>array([[0.01, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09, 0.1],
        [0.11, 0.12, 0.13, 0.14, 0.15, 0.16, 0.17, 0.18, 0.19, 0.2],
        [0.21, 0.22, 0.23, 0.24, 0.25, 0.26, 0.27, 0.28, 0.29, 0.3],
        [0.31, 0.32, 0.33, 0.34, 0.35, 0.36, 0.37, 0.38, 0.39, 0.4],
        [0.41, 0.42, 0.43, 0.44, 0.45, 0.46, 0.47, 0.48, 0.49, 0.5],
        [0.51, 0.52, 0.53, 0.54, 0.55, 0.56, 0.57, 0.58, 0.59, 0.6],
        [0.61, 0.62, 0.63, 0.64, 0.65, 0.66, 0.67, 0.68, 0.69, 0.7],
        [0.71, 0.72, 0.73, 0.74, 0.75, 0.76, 0.77, 0.78, 0.79, 0.8],
        [0.81, 0.82, 0.83, 0.84, 0.85, 0.86, 0.87, 0.88, 0.89, 0.9],
        [0.91, 0.92, 0.93, 0.94, 0.95, 0.96, 0.97, 0.98, 0.99, 1. ]])</code>

## Индексация Numpy

Здесь Вы будете работать с некоторой матрицей - чтобы создать её, запустите следующую ячейку! После этого мы попросим Вас воспроизвести несколько результатов работы с этой матрицей:

In [43]:
mat = np.arange(1, 26).reshape(5, 5)
mat

array([[ 1,  2,  3,  4,  5],
       [ 6,  7,  8,  9, 10],
       [11, 12, 13, 14, 15],
       [16, 17, 18, 19, 20],
       [21, 22, 23, 24, 25]])

#### 13. Напишите код, который на основе матрицы mat выведет указанную ниже матрицу (подсказка - используйте индексацию).

In [54]:
# Ваше решение тут
mat[2:, 1:]

array([[12, 13, 14, 15],
       [17, 18, 19, 20],
       [22, 23, 24, 25]])

#### 14. Напишите код, который на основе матрицы mat выведет указанную ниже значение.

In [46]:
# Ваше решение тут
mat[3, 4]

np.int64(20)

Результат:

<code>np.int64(20)</code>

#### 15. Напишите код, который на основе матрицы mat выведет указанную ниже матрицу.

In [57]:
# Ваше решение тут
mat[0:3, 1:2]

array([[ 2],
       [ 7],
       [12]])

Результат:

<code>array([[ 2],
        [ 7],
        [12]])</code>

#### 16. Напишите код, который на основе матрицы mat выведет указанную ниже вектор.

In [59]:
# Ваше решение тут
mat[4,:]

array([21, 22, 23, 24, 25])

Результат:

<code>array([21, 22, 23, 24, 25])</code>

#### 17. Напишите код, который на основе матрицы mat выведет указанную ниже матрицу.

In [60]:
# Ваше решение тут
mat[3:,:]

array([[16, 17, 18, 19, 20],
       [21, 22, 23, 24, 25]])

Результат:

<code>array([[16, 17, 18, 19, 20],
        [21, 22, 23, 24, 25]])</code>

## Операции в NumPy

#### 18. Найдите сумму всех чисел в матрице mat

In [62]:
# Ваше решение тут
mat.sum()

np.int64(325)

Результат:

<code>np.int64(325)</code>

#### 19. Найдите среднеквадратичное отклонение значений в матрице mat

In [63]:
# Ваше решение тут
mat.std()

np.float64(7.211102550927978)

Результат:

<code>np.float64(7.211102550927978)</code>

In [67]:
# Ваше решение тут
mat.prod(axis=0)

array([ 22176,  62832, 129168, 229824, 375000])

Результат:

<code>array([22176, 62832, 129168, 229824, 375000])</code>